src/test/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptorUnitTest.java

```diff
@@ -0,0 +1,326 @@
+                new Object(),
+                Collections.singletonList(new GATKAnnotationPluginDescriptor(null, null)),
+                Collections.emptySet());
+        String[] args = arguments.toArray(new String[arguments.size()]);
+        clp.parseArguments(nullMessageStream, args);
+        Collection<Annotation> annots = instantiateAnnotations(clp);
+    }
+
+    @DataProvider
+    public Object[][] annotationsWithRequiredArguments(){
+        return new Object[][]{{ InbreedingCoeff.class.getSimpleName(), "--founderID", "s1"}};
+    }
+
+    // fail if an annotation with required arguments is specified without corresponding arguments
+    // TODO this test is disabled as there are currently no annotations for which the argument is required
+    @Test(dataProvider = "annotationsWithRequiredArguments", expectedExceptions = CommandLineException.MissingArgument.class, enabled = false)
+    public void testDependentAnnotationArguments(
+            final String annot,
+            final String argName,   //unused
+            final String[] argValue) { //unused
+        CommandLineParser clp = new CommandLineArgumentParser(
+                new Object(),
+                Collections.singletonList(new GATKAnnotationPluginDescriptor(null, null)),
+                Collections.emptySet());
+
+        String[] args = {"--annotation", annot};  // no args, just enable filters
+        clp.parseArguments(nullMessageStream, args);
+    }
+
+    // fail if a annotation's arguments are passed but the annotation itself is not enabled
+    @Test(dataProvider = "annotationsWithRequiredArguments", expectedExceptions = CommandLineException.class)
+    public void testDanglingAnnotationArguments(
+            final String annot,
+            final String argName,
+            final String argValue) {
+        CommandLineParser clp = new CommandLineArgumentParser(
+                new Object(),
+                Collections.singletonList(new GATKAnnotationPluginDescriptor(null, null)),
+                Collections.emptySet());
+
+        String[] args = { argName, argValue }; // no annotation set
+
+        // no need to instantiate the annotation - dependency errors are caught by the command line parser
+        clp.parseArguments(nullMessageStream, args);
+    }
+
+    // Annotations with arguments to verify annotation test method actually filters
+    @DataProvider
+    public Object[][] annotationsWithGoodArguments(){
+        return new Object[][]{
+                { InbreedingCoeff.class.getSimpleName(), (Consumer<Annotation>)a -> {
+                            Assert.assertEquals(Double.valueOf((String) ((InbreedingCoeff) a)
+                                            .annotate(null, inbreedingCoefficientVC, null)
+                                            .get(GATKVCFConstants.INBREEDING_COEFFICIENT_KEY)),
+                                    -0.3333333, 0.001, "InbreedingCoefficientScores"); },
```

Comment:
Unfortunately, we don't have more than 1 annotation that includes arguments as of now. I should probably make mention of this feature somewhere so that people can implement some of these annotators for us.

Reply:
Even so, I still think we should untangle/refactor this data provider, or even get rid of it and just have the test method.